skiko/src/jvmMain/objectiveC/macos/Drawlayer.mm

```diff
@@ -69,88 +69,98 @@ -(void) dealloc
     NSView* zoomButtonView = [self.window standardWindowButton:NSWindowZoomButton];
     CGFloat horizontalButtonOffset = 20.0;
 
-    NSArray* windowButtons = @[ closeButtonView, miniaturizeButtonView, zoomButtonView ];
-    for (NSUInteger i = 0; i < windowButtons.count; i++) {
-        NSView* button = [windowButtons objectAtIndex:i];
+    [@[closeButtonView, miniaturizeButtonView, zoomButtonView] enumerateObjectsUsingBlock:^(NSView* button, NSUInteger i, BOOL* stop)
+    {
         button.translatesAutoresizingMaskIntoConstraints = NO;
-        [NSLayoutConstraint activateConstraints:@[
+        [newConstraints addObjectsFromArray:@[
             [button.centerYAnchor constraintEqualToAnchor:titlebar.centerYAnchor],
             [button.centerXAnchor constraintEqualToAnchor:titlebar.leftAnchor constant:(_customHeaderHeight/2.0 + (i * horizontalButtonOffset))],
         ]];
-    }
+    }];
+
+    [NSLayoutConstraint activateConstraints:newConstraints];
 }
 
 - (void) resetHeader
 {
+    NSView* themeFrame = self.window.contentView.superview;
     NSView* titlebarContainer = [self.window standardWindowButton:NSWindowCloseButton].superview.superview;
     NSView* titlebar = titlebarContainer.subviews[0];
     NSView* titlebarDecoration = titlebarContainer.subviews[1];
-    NSView* titlebarVisualEffect = titlebar.subviews[0];
-    NSView* titlebarBackground = titlebar.subviews[1];
+
+    // The following two views are only there on Big Sur and forward
+    BOOL runningAtLeastBigSur = [NSProcessInfo.processInfo isOperatingSystemAtLeastVersion:{ .majorVersion = 11, .minorVersion = 0, .patchVersion = 0 }];
+    NSView* titlebarVisualEffect = runningAtLeastBigSur ? titlebar.subviews[0] : nil;
+    NSView* titlebarBackground = runningAtLeastBigSur ? titlebar.subviews[1] : nil;
+
     NSView* closeButtonView = [self.window standardWindowButton:NSWindowCloseButton];
     NSView* miniaturizeButtonView = [self.window standardWindowButton:NSWindowMiniaturizeButton];
     NSView* zoomButtonView = [self.window standardWindowButton:NSWindowZoomButton];
 
-    NSArray* changedViews = @[titlebarContainer, titlebarDecoration, titlebar, titlebarVisualEffect, titlebarBackground, closeButtonView, miniaturizeButtonView, zoomButtonView];
+    NSArray* changedViews = runningAtLeastBigSur
+        ? @[titlebarContainer, titlebarDecoration, titlebar, titlebarVisualEffect, titlebarBackground, closeButtonView, miniaturizeButtonView, zoomButtonView]
+        : @[titlebarContainer, titlebarDecoration, titlebar, closeButtonView, miniaturizeButtonView, zoomButtonView];
     for (NSView* changedView in changedViews)
     {
         [changedView removeConstraints:changedView.constraints];
         changedView.translatesAutoresizingMaskIntoConstraints = YES;
     }
-    NSView* dragger = titlebar.subviews[1].subviews[0];
+    NSView* dragger = titlebar.subviews[titlebar.subviews.count - 1];
```

Comment:
Please add check that `titlebar.subviews.count > 0`

Reply:
Done!